test/connectivity/scripts/connectivityTest.sh

```diff
@@ -42,6 +42,11 @@ function prepare_test_from_artifacts() {
     sed -i -e "s@<Container_Registry>@$CONTAINER_REGISTRY@g" "$deployment_working_file"
```

Comment:
In general, I think we should refactor this script. The problem previously we had in runE2ETests.sh is that we combine all the scenario into 1 single script file and therefore the input parameters are union of all scenarios.  It makes difficult to maintain.

To solve this problem, I started to create a testHelper.sh to extract common method/function for reuse.  What I would suggest:
1. Extract everything in connectivityTest.sh into small functions.  For example, process_args should be broken down into many methods, e.g. process_args_for_basic_parameters, process_args_for_metric, process_args_for_customBuild, etc; then move all these methods into testHelper.sh.  And move testHelper.sh to test folder for reuse.
2. create a new folder "benchmark" under test folder.
3. create a new "benchmark.sh" to use testHelper.sh to build the tests and uses exactly what that test really needs.  For example, I expect it will call prepare_test_from_artifacts_for_basic, prepare_test_from_artifacts_for_protocolHeads, but not prepare_test_from_artifacts_for_TwinTests.
4. With this new test script, a new yaml "benchmark-deploy.yaml" should be created.

Reply:
This is not just to make your tests to work; this will help us to easily maintain our test script.  Check with Varun if he would like to do this refactoring later.  This is trade-off between code maintenance and deadline.

Reply:
I think this should be a task of its own. Would it be alright if I just do a setup of the pipeline + update the deployment file then I'll create a task to re-factor the bash files later?

Reply:
Please confirm with Varun.

Reply:
I created this task to follow up.
https://msazure.visualstudio.com/One/_workitems/edit/6953217